consensus/spos/bls/subroundEndRound.go

```diff
@@ -621,7 +680,9 @@ func (sr *subroundEndRound) getFullMessagesForInvalidSigners(invalidPubKeys []st
 }
 
 func (sr *subroundEndRound) handleInvalidSignersOnAggSigFail() ([]byte, []byte, error) {
-	invalidPubKeys, err := sr.verifyNodesOnAggSigFail()
+	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
```suggestion
	ctx, cancel := context.WithCancel(context.Background())
```

Reply:
with this implementation, I don't think timeout will ever be reached, perhaps the node can get stuck
reason: 
- cancel is called when verifyNodesOnAggSigFail is finished
- verifyNodesOnAggSigFail has a blocking call to checkGoRoutinesThrottler
- checkGoRoutinesThrottler ends in 2 conditions:
   - the signature can be sent
   - timeout

maybe a better approach here would be to use a context with timeout = with round duration:
```go
ctx, cancel := context.WithTimeout(context.Background(), sr.RoundHandler().TimeDuration())
```